zhaquirks/sonoff/swv.py

```diff
@@ -28,13 +31,30 @@ class AttributeDefs(BaseAttributeDefs):
             type=ValveState,
         )
 
-    @property
-    def _is_manuf_specific(self):
-        return False
-
 
 (
     QuirkBuilder("SONOFF", "SWV")
-    .replaces(EwelinkCluster)
+    .replaces(CustomSonoffCluster)
+    .binary_sensor(
+        CustomSonoffCluster.AttributeDefs.water_valve_state.name,
+        CustomSonoffCluster.cluster_id,
+        device_class=BinarySensorDeviceClass.PROBLEM,
```

Comment:
Ah, actually. Can we use `BinarySensorDeviceClass.MOISTURE` here? Does that make sense for this device?

That's the device class we use for (IAS Zone) leak sensors in ZHA. If that works, we could also drop the `translation_key` if the default name ("Moisture") is clear enough.

Reply:
"Moisture" for leak seems odd IMO, but I will give it a try and report back.

Reply:
Leak sensors generally use `BinarySensorDeviceClass.MOISTURE` [in HA](https://github.com/search?q=repo%3Ahome-assistant%2Fcore%20BinarySensorDeviceClass.MOISTURE&type=code).
The entity icon will show as a water droplet (crossed through or not) and be "dry" or "wet".
I'm not sure where a leak sensor is positioned on this device, but I think the device class is fitting here.

Reply:
I agree that `MOISTURE` is the proper class here. Also [from the docs](https://developers.home-assistant.io/docs/core/entity/binary-sensor/): `On means wet, Off means dry.`

Reply:
Will apply this suggestion to get it merged:
```suggestion
        device_class=BinarySensorDeviceClass.MOISTURE,
```